code/optimize/src/test/java/com/adobe/marketing/mobile/optimize/OptimizeExtensionTests.java

```diff
@@ -1200,6 +1203,56 @@ public void testHandleEdgeResponse_requestEventIdNotBeingTracked() throws Except
         }
     }
 
+    @Test
+    public void testHandleEdgeResponse_missingItemData() throws IOException {
+        // Setup
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String, Object> edgeResponseData =
+                objectMapper.readValue(
+                        Objects.requireNonNull(getClass().getClassLoader())
+                                .getResource(
+                                        "json/EVENT_DATA_EDGE_RESPONSE_MISSING_ITEM_DATA.json"),
+                        HashMap.class);
+
+        Event testEvent =
+                new Event.Builder(
+                                "AEP Response Event Handle",
+                                "com.adobe.eventType.edge",
+                                "personalization:decisions")
+                        .setEventData(edgeResponseData)
+                        .build();
+
+        ArgumentCaptor<Event> eventCaptor = ArgumentCaptor.forClass(Event.class);
+        extension.setUpdateRequestEventIdsInProgress(
+                "AAAAAAAA-AAAA-AAAA-AAAA-AAAAAAAAAAAA",
+                new ArrayList<DecisionScope>() {
+                    {
+                        add(new DecisionScope("someDecisionScope"));
+                    }
+                });
+        extension.handleEdgeResponse(testEvent);
+        Mockito.verify(mockExtensionApi, Mockito.times(1)).dispatch(eventCaptor.capture());
+
+        List<Map<String, Object>> propositionsList =
+                (List<Map<String, Object>>)
+                        eventCaptor.getValue().getEventData().get("propositions");
+
+        Assert.assertNotNull("Propositions list should not be null", propositionsList);
+        try {
+            List<Map<String, Object>> items =
+                    DataReader.getTypedListOfMap(
+                            Object.class,
+                            propositionsList.get(0),
+                            OptimizeConstants.JsonKeys.PAYLOAD_ITEMS);
+            Assert.assertNotNull("Items list should not be null", items);
+            Assert.assertNotNull(
+                    "Item data should not be null",
+                    items.get(0).get(OptimizeConstants.JsonKeys.PAYLOAD_ITEM_DATA));
```

Comment:
Please add assertions to verify the `Offer` details. You can refer to existing `testHandleEdgeResponse_validProposition` for more details

Reply:
done @spoorthipujariadobe